### ISSUE TITLE
view::adjacent_difference wanted

### ISSUE THREAD

**Issue by lichray** [ericniebler/range-v3#243]
I encountered a need to compute the minimal adjacent difference.

**Comment by ericniebler** [comment]
Patches welcome. :-)

**Comment by tivek** [comment]
Wouldn't something quick like this work?
```C++
template<typename Rng>
auto adj_difference(Rng r)
{
    return view::transform(r, r | view::drop(1), [](auto a, auto b) {
        return b - a;
    });
}
auto min_diff = ranges::min(adj_difference(someview));
```

**Comment by lichray** [comment]
Yes, but first this is harder to understand and harder to write than a raw loop, second it's iterating on two ranges.

Additionally I consider both this and `adjacent_difference` (copying first element, limiting the functor's return type to be the element type) to be useful.  Numpy has both, `diff` and `ediff1d`.  Maybe we can also have `view::difference`.
